### PR TITLE
Delete FunctionParse copy constructor.

### PR DESCRIPTION
**Pull request by bangerth**
In response to #8951. This just makes more obvious what is already the case.

### CODE REVIEW

**Comment by bangerth** [comment]
@rezarastak -- FYI.

File: include/deal.II/base/function_parser.h [hunk 1]
```diff
@@ -246,6 +246,12 @@ class FunctionParser : public AutoDerivativeFunction<dim>
                                                      ",t",
                  const double h = 1e-8);
 
+  /**
+   * Copy constructor. Objects of this type can not be copied, and
+   * consequently this constructor is deleted.
+   */
+  FunctionParse(const FunctionParser &) = delete;
```
Comment (masterleinad) [inline include/deal.II/base/function_parser.h]:
```suggestion
  FunctionParser(const FunctionParser &) = delete;
```

Comment (bangerth) [reply include/deal.II/base/function_parser.h]: That's what happens if you're working on an unplugged laptop and don't want to compile the entire library...

**Comment by bangerth** [comment]
I've now also made the copy operator `delete`d as that of course had the same problem. Let me think about move constructors and operators -- I think they could work, but if you allow me, I'll take care of that in a separate patch.

/rebuild

**Comment by masterleinad** [comment]
Sure. Defining the copy constructor implies deleting the move constructor so you are already touching that implicitly.

**Comment by bangerth** [comment]
It turns out that some of the TBB data structures we use in this class have no appropriate move constructors/operators, so the compiler falls back to the copy constructors/operators instead that then, however, turns out to be deleted. In other words, moving doesn't work for these kinds of objects -- and so I've deleted the move operators to make that clear.

**Pull request merged**